sql/core/src/test/resources/sql-tests/inputs/postgreSQL/limit.sql

```diff
@@ -0,0 +1,164 @@
+-- fetch all in c3;
+-- fetch 1 in c3;
+-- fetch backward 1 in c3;
+-- fetch backward all in c3;
+-- fetch backward 1 in c3;
+-- fetch all in c3;
+--
+-- declare c4 cursor for select * from int8_tbl offset 10;
+-- fetch all in c4;
+-- fetch 1 in c4;
+-- fetch backward 1 in c4;
+-- fetch backward all in c4;
+-- fetch backward 1 in c4;
+-- fetch all in c4;
+--
+-- rollback;
+
+DROP VIEW INT8_TBL;
+
+-- Stress test for variable LIMIT in conjunction with bounded-heap sorting
+
+-- [SPARK-28330] ANSI SQL: Top-level <result offset clause> in <query expression>
+-- SELECT
+--   (SELECT n
+--      FROM (VALUES (1)) AS x,
+--           (SELECT n FROM generate_series(1,10) AS n
+--              ORDER BY n LIMIT 1 OFFSET s-1) AS y) AS z
+--   FROM generate_series(1,10) AS s;
+
+--
+-- Test behavior of volatile and set-returning functions in conjunction
+-- with ORDER BY and LIMIT.
+--
+
+-- [SPARK-29631] Support ANSI SQL CREATE SEQUENCE
+-- create temp sequence testseq;
+
+-- explain (verbose, costs off)
+-- select unique1, unique2, nextval('testseq')
+--   from tenk1 order by unique2 limit 10;
+
+-- select unique1, unique2, nextval('testseq')
+--   from tenk1 order by unique2 limit 10;
+
+-- select currval('testseq');
+
+-- explain (verbose, costs off)
+-- select unique1, unique2, nextval('testseq')
+--   from tenk1 order by tenthous limit 10;
+
+-- select unique1, unique2, nextval('testseq')
+--   from tenk1 order by tenthous limit 10;
+
+-- select currval('testseq');
+
+-- explain (verbose, costs off)
+-- select unique1, unique2, generate_series(1,10)
+--   from tenk1 order by unique2 limit 7;
+
+-- [SPARK-27767] Built-in function: generate_series
+-- select unique1, unique2, generate_series(1,10)
+--   from tenk1 order by unique2 limit 7;
+
+-- explain (verbose, costs off)
+-- select unique1, unique2, generate_series(1,10)
+--   from tenk1 order by tenthous limit 7;
+
+-- [SPARK-27767] Built-in function: generate_series
+-- select unique1, unique2, generate_series(1,10)
+--   from tenk1 order by tenthous limit 7;
+
+-- use of random() is to keep planner from folding the expressions together
+-- explain (verbose, costs off)
+-- select generate_series(0,2) as s1, generate_series((random()*.1)::int,2) as s2;
+
+-- [SPARK-27767] Built-in function: generate_series
+-- select generate_series(0,2) as s1, generate_series((random()*.1)::int,2) as s2;
+
+-- explain (verbose, costs off)
+-- select generate_series(0,2) as s1, generate_series((random()*.1)::int,2) as s2
+-- order by s2 desc;
+
+-- [SPARK-27767] Built-in function: generate_series
+-- select generate_series(0,2) as s1, generate_series((random()*.1)::int,2) as s2
+-- order by s2 desc;
+
+-- test for failure to set all aggregates' aggtranstype
+-- explain (verbose, costs off)
+-- select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
+--   from tenk1 group by thousand order by thousand limit 3;
+
+select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
```

Comment:
Although I believe `Apache Spark` behavior `random()*0` == `0.0` is more natural, do we have a JIRA issue for this difference from PostgreSQL where `random()*0` == `0`?

Reply:
Sorry, but I missed your point and the result seems to be the same;
```
select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
  from tenk1 group by thousand order by thousand limit 3;
  s1   |  s2   
-------+-------
 45000 | 45000
 45010 | 45010
 45020 | 45020
(3 rows)
```
https://github.com/postgres/postgres/blob/REL_12_STABLE/src/test/regress/expected/limit.out#L497-L504

Also, I tried below;
```
postgres=# select * from (values (1)) t(v) where random() * 0.0 = 0.0;
 v 
---
 1
(1 row)

scala> sql("""select * from (values (1)) t(v) where random() * 0.0 = 0.0""").show()
+---+
|  v|
+---+
|  1|
+---+
```
What's the difference here?

Reply:
The generated result is not the same in this PR.